execution.go

```diff
@@ -28,7 +29,62 @@ func SetupExecutionEnvironment() {
 	fset = token.NewFileSet()
 }
 
-// OutputMsg holds the data for a pyout message.
+type KernelStatusMsg struct {
+	ExecState string `json:"execution_state"`
+}
+
+func setBusy(receipt MsgReceipt) {
+	var status KernelStatusMsg
+
+	msg := NewMsg("status", receipt.Msg)
+	status.ExecState = "busy"
+	msg.Content = status
+
+	receipt.SendResponse(receipt.Sockets.IOPubSocket, msg)
+}
+
+func setIdle(receipt MsgReceipt) {
+	var status KernelStatusMsg
+
+	msg := NewMsg("status", receipt.Msg)
+	status.ExecState = "idle"
+	msg.Content = status
+
+	receipt.SendResponse(receipt.Sockets.IOPubSocket, msg)
+}
+
+
+type ExecuteInputMsg struct {
+	ExecCount int    `json:"execution_count"`
+	Code      string `json:"code"`
+}
+
+func publishExecuteInput(receipt MsgReceipt, execCount int, code string) {
+	var pubExecInputContent ExecuteInputMsg
```

Comment:
not sure I understand why you create this `ExecuteInputMsg`, set 2 fields and then essentially drop it on the floor.
isn't there some typo somewhere or a line like the following that's missing:

```go
pubExecInput.Content = pubExecInputContent
```

if that's the case, I would just rename `pubExecInputContent` into `content`.
actually, I would just forgo the declaration of that variable, and thus write:

```go
pubExecInput.Content = ExecuteInputMsg{ExecCount: execCount, Code: code}
```